.githooks/pre-commit

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+ENTERPRISE_COMMIT_FILE=.enterprise-commit
+ENTERPRISE_REPO=./enterprise
+
+# Check enterprise repo is loaded
+if [ ! -d "$ENTERPRISE_REPO" ]; then
+  exit 0
+fi
+
+# check if enterprise repo has any unstaged changes
+ENTERPRISE_UNSTAGED=`cd $ENTERPRISE_REPO; git status -s -uall`
+if [ ! -z "$ENTERPRISE_UNSTAGED" ]; then
+  echo "ERROR: There are unstaged changes in enterprise repo. Please, commit and push them first."
+  echo "git status of $ENTERPRISE_RERO"
+  (cd $ENTERPRISE_REPO; git status)
+  exit 1
+fi
+
+# check if enterprise repo has no remote
+ENTERPRISE_BRANCH=`cd $ENTERPRISE_REPO; git name-rev --name-only HEAD`
+ENTERPRISE_REMOTE=`cd $ENTERPRISE_REPO; git config branch.$ENTERPRISE_BRANCH.remote`
+if [ -z "$ENTERPRISE_REMOTE" ]; then
+  echo "ERROR: Current enterprise repo branch '$ENTERPRISE_BRANCH' has no remote. Please, push your enterprise changes first."
+  exit 1
+fi
+
+ENTERPRISE_REMOTE_EXISTS=`cd $ENTERPRISE_REPO; git branch -a | grep remotes/$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH`
+if [ -z "$ENTERPRISE_REMOTE_EXISTS" ]; then
+  echo "ERROR: Enterprise repo remote branch '$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH' does not yet exist. Please, push your enterprise changes first."
+  exit 1
+fi
+
+# check if enterprise repo has unpushed commits
+ENTERPRISE_DIFF=`cd $ENTERPRISE_REPO; git rev-list --left-right $ENTERPRISE_BRANCH...$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH`
+
+if [ ! -z "$ENTERPRISE_DIFF" ]; then
+  echo "ERROR: There are either unpushed changes, or current branch '$ENTERPRISE_BRANCH' is behind '$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH'. Please, update your branch, commit and push your changes."
```

Comment:
I realise that it might be better to allow the user to proceed, after presenting a warning, than stopping them completely. This way you get both protection and flexibility.